### DISCUSSION TITLE
Graceful shutdown with Chrome does not happen until second request is received

### DISCUSSION THREAD

**Discussion by surya-prakash-susarla** [hyperium/hyper#3095] [Q&A]
Hey everyone, I am trying to create a server which gracefully shuts down after a single request has been received. The implementation that I have right now works but I ran into an interesting issue which I see others have encountered in the past. The server does not shutdown although the signal is fired after the request is handled. It only happens when the request comes from Chrome and the second request in such scenario ends up getting served and leads to a shutdown of the server. When I try to do the same from Safari this does not happen and everything happens as expected i.e., server shuts down after serving the first request. Is there something I can do to resolve this issue?

Thanks!

**Comment by Cloud33** [comment]
I also encounter this problem. If there is no second request, it takes a very long time to shut down https://github.com/hyperium/hyper/issues/3188
